src/main/java/org/zalando/nakadi/service/subscription/zk/CuratorZkSubscriptionClient.java

```diff
@@ -317,4 +326,101 @@ public ZkSubscriptionNode getZkSubscriptionNodeLocked() {
 
         return subscriptionNode;
     }
+
+    @Override
+    public ZKSubscription subscribeForCursorsReset(final Runnable listener)
+            throws NakadiRuntimeException, UnsupportedOperationException {
+        final NodeCache cursorResetCache = new NodeCache(curatorFramework, resetCursorPath);
+        cursorResetCache.getListenable().addListener(() -> listener.run());
+
+        try {
+            cursorResetCache.start();
+        } catch (final Exception e) {
+            throw new NakadiRuntimeException(e);
+        }
+
+        return new ZKSubscription() {
+            @Override
+            public void refresh() {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void cancel() {
+                try {
+                    cursorResetCache.getListenable().clear();
+                    cursorResetCache.close();
+                } catch (final IOException e) {
+                    throw new NakadiRuntimeException(e);
+                }
+            }
+        };
+    }
+
+    @Override
+    public boolean isCursorResetInProgress() {
+        try {
+            return curatorFramework.checkExists().forPath(resetCursorPath) != null;
+        } catch (final Exception e) {
```

Comment:
I do not think it has much sense, because we will have a lot exceptions in our logs, which mean nothing to us